### ISSUE TITLE
Non-ASCII characters in testsets and report JSON files

### ISSUE THREAD

**Issue by PierreMesure** [Giskard-AI/giskard#2046]
### Issue Type

Bug

### Source

source

### Giskard Library Version

2.15.1

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.12

### Current Behaviour?

```shell
Right now, the testsets are generated with ASCII characters in the texts. This is not optimal with content in Swedish, for instance.
```

### Standalone code OR list down the steps to reproduce the issue

```shell
Generate a testset in Swedish and/or a report.
```

**Comment by PierreMesure** [comment]
My suggestion is to add `force_ascii=False` in these two calls to Pandas `to_json()` function: [testset](https://github.com/Giskard-AI/giskard/blob/750db5c8f87193118d64d71ccbfe513e5132ecb7/giskard/rag/testset.py#L99), [report](https://github.com/Giskard-AI/giskard/blob/750db5c8f87193118d64d71ccbfe513e5132ecb7/giskard/rag/report.py#L161). I can send a PR if you agree.

**Comment by henchaves** [comment]
Hey @PierreMesure, thanks for opening this issue.
Yes, of course, you can open a PR and request my review

**Comment by PierreMesure** [comment]
I did it, let me know if you want a test as well. It might take more time.

**Comment by henchaves** [comment]
Thanks @PierreMesure!
Yes, please. Could you add a simple test, for example saving and loading a testset in Swedish, to make sure that the content of the loaded one is similar to the original one?
Thanks a lot!

**Comment by PierreMesure** [comment]
I'll try! I also noticed that non ASCII-characters end up in other JSON files when generated with [json.dumps](https://github.com/Giskard-AI/giskard/blob/750db5c8f87193118d64d71ccbfe513e5132ecb7/giskard/rag/report.py#L166) so I'll fix it as wel.

**Comment by henchaves** [comment]
Thanks a lot @PierreMesure

**Comment by PierreMesure** [comment]
Done now. I only added `ensure_ascii=False` in [report.py](https://github.com/Giskard-AI/giskard/blob/750db5c8f87193118d64d71ccbfe513e5132ecb7/giskard/rag/report.py) so you might want to add it in other places where JSON files are saved. But I didn't want to create too much disruption.

I also added a test inspired by another one I found in the same test file but I couldn't get Pytest to work so I don't know if it works. 🤔
